### ISSUE TITLE
feat: 画廊功能建议

### ISSUE THREAD

**Issue by MiayS** [zfile-dev/zfile#535]
4.x版本的画廊能不能改成像3.x版本的图片模式一样，目录下没有图片会显示目录，有图片画廊自动列出图片

**Comment by zhaojun1998** [comment]
没图片怎么显示目录，尤其在多栏情况下， 还有既有图片，又有文件夹，文件如何显示，图片占得宽度可比普通图标大

**Comment by DeaneChen** [comment]
我猜测MiaoYas1的意思是，页面会根据当前目录下的文件情况判断自动处于目录显示模式或画廊模式。例如：当均为图片时默认为画廊模式，当存在其他文件或目录时则处于默认模式。

**Comment by vvnocode** [comment]
同意。3.x的图片模式会显示目录，能方便快速回到上一层或者去其他层。但是画廊模式只显示图片，体验差了很多。建议恢复3.x的图片模式

**Comment by tme1946** [comment]
其实我希望，如果设为默认画廊模式，可以遍历子文件夹下的所有图片

**Comment by tme1946** [comment]
> 其实我希望，如果设为默认画廊模式，可以遍历子文件夹下的所有图片
比如我的目录格式是：
photos
-北京
--故宫
--香山
-天津
如果我想作为画廊展示的话，会将photos设为存储源，并开启默认画廊模式

**Comment by 54tv** [comment]
> 其实我希望，如果设为默认画廊模式，可以遍历子文件夹下的所有图片

对  我也希望这个模式  因为大多人存图都是按文件夹来的  
另外画廊模式 现在不支持视频显示  如果以后能加上就更好了  毕竟照片视频经常在一起
